src/ITensorAutoHOOT/zygote.jl

```diff
@@ -0,0 +1,125 @@
+using Zygote, AutoHOOT
+using Zygote: @adjoint
+using ITensors: setinds
+
+const ad = AutoHOOT.autodiff
+const go = AutoHOOT.graphops
+
+# represent a sum of tensor networks
+mutable struct NetworkSum
+  nodes::Array
+end
+
+struct Executor
+  net_sums::Array{<:NetworkSum}
+  feed_dict::Dict
+end
+
+NetworkSum() = NetworkSum([])
+
+function Executor(networks::Array)
+  nodes, node_dict = generate_einsum_expr(networks)
+  # TODO: add caching here
+  net_sums = [NetworkSum([go.generate_optimal_tree(n)]) for n in nodes]
+  return Executor(net_sums, node_dict)
+end
+
+@adjoint Executor(networks::Array) = Executor(networks), dv -> (nothing,)
```

Comment:
Note that you can use the `@non_differentiable` macro defined in `ChainRulesCore` (https://juliadiff.org/ChainRulesCore.jl/stable/writing_good_rules.html#[@non_differentiable](@ref)).

Reply:
Done